### PR TITLE
chore: Use `value::Value` directly in the codebase

### PR DESCRIPTION
**Pull request by blt**
This commit modifies the project to use `value::Value` directly and not as a
re-export from VRL. The changeset is large but almost entirely mechanical. I
would like to experiment with different representations of `Value` -- imagine a
world where the `Value` is mostly stack allocated, or where the `Value::Object`
doesn't recurse to another heap allocated `Value` outside of the current virtual
machine page -- but I realized while making some modifications to that type that
its representation was, well, kindly spread through the project.

We don't solve that here. Instead we take a single step in that
direction. Follow-up work will work a struct shim over the top of the `BTreeMap`
inside `Value` so callers are not aware of the `BTreeMap`. That abstraction is
what will allow us to fiddle with the representation, so long as we hew to the
enum nature of `Value`.

Signed-off-by: Brian L. Troutwine <brian@troutwine.us>

### CODE REVIEW

**Comment by JeanMertz** [comment]
> We don't solve that here. Instead we take a single step in that
> direction. Follow-up work will work a struct shim over the top of the `BTreeMap`
> inside `Value` so callers are not aware of the `BTreeMap`. That abstraction is
> what will allow us to fiddle with the representation, so long as we hew to the
> enum nature of `Value`.

I don't quite understand how this work is blocked by having `Value` be re-exported in VRL? That is, adding such a shim would automatically propagate to all uses of `Value`, regardless of direct or re-exported use, right?

**Comment by blt** [comment]
> > We don't solve that here. Instead we take a single step in that
> 
> > direction. Follow-up work will work a struct shim over the top of the `BTreeMap`
> 
> > inside `Value` so callers are not aware of the `BTreeMap`. That abstraction is
> 
> > what will allow us to fiddle with the representation, so long as we hew to the
> 
> > enum nature of `Value`.
> 
> 
> 
> I don't quite understand how this work is blocked by having `Value` be re-exported in VRL? That is, adding such a shim would automatically propagate to all uses of `Value`, regardless of direct or re-exported use, right?

That’s how I started out on this line of work. Practically the story is muddier, specifically in crates that use a mixture of types directly ‘value’ and those re-exported from ‘vrl’ where derivation comes into play. Enrichment crate especially had a serious serde spew that doesn’t exist with the approach taken here. An alternative might be to preserve the prelude reexport but crate private, that is, for VRL private use only. 

So, not blocked per se but the compiler output is more feasible to work with.

**Comment by JeanMertz** [comment]
> An alternative might be to preserve the prelude reexport but crate private, that is, for VRL private use only.
> 
> So, not blocked per se but the compiler output is more feasible to work with.

It's not a huge deal. My main concern would be to keep VRL function development for external contributors as simple as possible, which is why I added the prelude. If we could keep that as-is, that'd be great. But if not, it's not a big problem, as most people start by copying an existing function anyway, and those will have `Value` imported already.

**Review by JeanMertz — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -113,7 +113,7 @@ lookup = { path = "lib/lookup" }
 portpicker = { path = "lib/portpicker" }
 prometheus-parser = { path = "lib/prometheus-parser", optional = true }
 tracing-limit = { path = "lib/tracing-limit" }
-value = { path = "lib/value", optional = true }
+value = { path = "lib/value", optional = false }
```
Comment (bruceg) [inline Cargo.toml:116]: Do you need the `optional = false`? I thought that was the default.

File: src/internal_events/kubernetes/instrumenting_watcher.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+use std::fmt::Debug;
```
Comment (bruceg) [inline src/internal_events/kubernetes/instrumenting_watcher.rs:1]: Did this file need to be reintroduced, or is this an accidental re-add?

File: Cargo.toml (hunk 1 above)
Comment (blt) [reply Cargo.toml:116]: It is. I'll remove this.

File: src/internal_events/kubernetes/instrumenting_watcher.rs (hunk 2 above)
Comment (blt) [reply src/internal_events/kubernetes/instrumenting_watcher.rs:1]: Hrm. Rebase goofiness. Thank you for catching this.

**Review by bruceg — APPROVED** [review]

**Comment by blt** [comment]
That’s how I started out on thIs line of work. Practically the story is muddier, specifically in crates that use a mixture of types directly ‘value’ and those re-exported from ‘vrl’ where derivation comes into play. Enrichment crate especially had a serious serde spew that doesn’t exist with the approach taken here. An alternative might be to have VRL re-export but at the crate level only, that is, for VRL private use. 

So, not blocked per se but the compiler output is more feasible to work with.

**Pull request merged**